server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -138,12 +138,17 @@ public synchronized void updateSegments(final SegmentInfos infos) throws IOExcep
             ensureOpen();
             final long maxSeqNo = Long.parseLong(infos.userData.get(MAX_SEQ_NO));
             final long incomingGeneration = infos.getGeneration();
-            readerManager.updateSegments(infos);
+            boolean remoteStoreEnabled = engineConfig.getIndexSettings().isRemoteStoreEnabled();
```

Comment:
I think we should set this as a "shouldCommit" field on the engine thats set in our constructor.